lib/oli_web/live/workspaces/course_author/editor_live.ex

```diff
@@ -117,61 +125,79 @@ defmodule OliWeb.Workspaces.CourseAuthor.Curriculum.EditorLive do
     part_component_types = PartComponents.part_components_for_project(project)
 
+    breadcrumbs =
+      Breadcrumb.trail_to(project_slug, revision_slug, AuthoringResolver, project.customizations)
+
     content = %{
       active: :curriculum,
       activity_types: activity_types,
-      breadcrumbs:
-        Breadcrumb.trail_to(
-          project_slug,
-          revision_slug,
-          Oli.Publishing.AuthoringResolver,
-          project.customizations
-        ),
+      breadcrumbs: breadcrumbs,
       collab_space_config: context.collab_space_config,
       graded: context.graded,
       is_admin?: is_admin?,
       part_component_types: part_component_types,
-      part_scripts: PartComponents.get_part_component_scripts(:authoring_script),
+      part_scripts: PartComponents.get_part_component_scripts(),
       project_slug: project_slug,
       context: context,
       raw_context: context,
       revision_slug: revision_slug,
-      scripts: Activities.get_activity_scripts(:authoring_script),
+      scripts: Activities.get_activity_scripts(),
       title: "Edit | " <> context.title,
       resource_title: project.title,
       resource_slug: project.slug
     }
 
-    content =
+    {content, target_scripts} =
       case context do
         %{content: %{"advancedAuthoring" => true}} ->
-          Map.put(content, :app_params, %{
-            isAdmin: is_admin?,
-            revisionSlug: revision_slug,
-            projectSlug: project_slug,
-            graded: context.graded,
-            content: context,
-            paths: %{
-              images: Routes.static_path(socket, "/images")
-            },
-            activityTypes: activity_types,
-            partComponentTypes: part_component_types,
-            appsignalKey: Application.get_env(:appsignal, :client_key),
-            initialSidebarExpanded: socket.assigns[:sidebar_expanded]
-          })
+          activity_type_scripts =
+            Enum.reduce(activity_types, [], fn %{slug: slug, authoring_script: authoring_script},
+                                               acc ->
+              if slug == "oli_adaptive", do: [authoring_script | acc], else: acc
+            end)
+
+          updated_content =
+            Map.put(content, :app_params, %{
+              isAdmin: is_admin?,
+              revisionSlug: revision_slug,
+              projectSlug: project_slug,
+              graded: context.graded,
+              content: context,
+              paths: %{images: Routes.static_path(socket, "/images")},
+              activityTypes: activity_types,
+              partComponentTypes: part_component_types,
+              appsignalKey: Application.get_env(:appsignal, :client_key),
+              initialSidebarExpanded: socket.assigns[:sidebar_expanded]
+            })
+
+          {updated_content, ["authoring.js"] ++ activity_type_scripts}
 
         _ ->
-          content
+          {content, ["pageeditor.js"]}
```

Comment:
This section is quite clumsy. It would be great to refactor it in a future iteration.